src/Dofs/ConstraintHandler.jl

```diff
@@ -913,13 +913,13 @@ coordinate axes.
 
 See the manual section on [Periodic boundary conditions](@ref) for more information.
 """
-struct PeriodicDirichlet
+struct PeriodicDirichlet{T}
     field_name::Symbol
     components::Vector{Int} # components of the field
     face_pairs::Vector{Pair{String,String}} # legacy that will populate face_map on add!
     face_map::Vector{PeriodicFacePair}
     func::Union{Function,Nothing}
-    rotation_matrix::Union{Matrix{Float64},Nothing}
+    rotation_matrix::Union{Matrix{T},Nothing}
 end
```

Comment:
I tried to not touch the ConstraintHandler (construction) logic, because I really don't want to break any assumption here -- I don't have an overview of what all the assumptions taken are. Hence I would leave refining/cleaning up everything related to the ConstraintHandler to future PRs.

Reply:
@fredrikekre after thinking more about this I think it makes sense to have two different structs for the constraint.  But I would leave this as future work, as this PR here is "only" concerned with paving the way for Float32 support (and possibly more types)

Reply:
Resolved / reverted in https://github.com/Ferrite-FEM/Ferrite.jl/pull/802/commits/1a097538b39ef3e32c9d175c8cfc493c5b38cd24